Tests/IntegrationTests/PropagationAtlasConstant.cpp

```diff
@@ -0,0 +1,175 @@
+                     phi, theta, p, q, s, bz) {
+  runToSurfaceTest(makePropagator(bz), geoCtx, magCtx,
+                   makeParametersCurvilinear(phi, theta, p, q), s,
+                   ZCylinderSurfaceBuilder(), epsPos, epsDir, epsMom,
+                   showDebug);
+}
+
+BOOST_DATA_TEST_CASE(ToDisc,
+                     ds::phi* ds::theta* ds::absMomentum* ds::chargeNonZero*
+                         ds::pathLength* ds::magneticField,
+                     phi, theta, p, q, s, bz) {
+  runToSurfaceTest(makePropagator(bz), geoCtx, magCtx,
+                   makeParametersCurvilinear(phi, theta, p, q), s,
+                   DiscSurfaceBuilder(), epsPos, epsDir, epsMom, showDebug);
+}
+
+BOOST_DATA_TEST_CASE(ToPlane,
+                     ds::phi* ds::theta* ds::absMomentum* ds::chargeNonZero*
+                         ds::pathLength* ds::magneticField,
+                     phi, theta, p, q, s, bz) {
+  runToSurfaceTest(makePropagator(bz), geoCtx, magCtx,
+                   makeParametersCurvilinear(phi, theta, p, q), s,
+                   PlaneSurfaceBuilder(), epsPos, epsDir, epsMom, showDebug);
+}
+
+// True forward/backward tracks do not work with z straws
+BOOST_DATA_TEST_CASE(ToStrawAlongZ,
+                     ds::phi* ds::thetaWithoutBeam* ds::absMomentum*
+                         ds::chargeNonZero* ds::pathLength* ds::magneticField,
+                     phi, theta, p, q, s, bz) {
+  runToSurfaceTest(makePropagator(bz), geoCtx, magCtx,
+                   makeParametersCurvilinear(phi, theta, p, q), s,
+                   ZStrawSurfaceBuilder(), epsPos, epsDir, epsMom, showDebug);
+}
+
+// check covariance transport using the ridders propagator for comparison
+
+BOOST_DATA_TEST_CASE(CovarianceCurvilinear,
+                     ds::phi* ds::theta* ds::absMomentum* ds::chargeNonZero*
+                         ds::pathLength* ds::magneticField,
+                     phi, theta, p, q, s, bz) {
+  runFreePropagationComparisonTest(
+      makePropagator(bz), makeRiddersPropagator(bz), geoCtx, magCtx,
+      makeParametersCurvilinearWithCovariance(phi, theta, p, q), s, epsPos,
+      epsDir, epsMom, epsCov, showDebug);
+}
+
+BOOST_DATA_TEST_CASE(
+    CovarianceToCylinderAlongZ,
+    ds::phiWithoutAmbiguity* ds::thetaWithoutBeam* ds::absMomentum*
+        ds::chargeNonZero* ds::pathLength* ds::magneticField,
+    phi, theta, p, q, s, bz) {
+  runToSurfaceComparisonTest(
+      makePropagator(bz), makeRiddersPropagator(bz), geoCtx, magCtx,
+      makeParametersCurvilinearWithCovariance(phi, theta, p, q), s,
+      ZCylinderSurfaceBuilder(), epsPos, epsDir, epsMom, epsCov, showDebug);
+}
+
+// NOTE msmk 2020-08-13
+// i suspect that this does not work because the Ridders propagator does not now
+// about the parameter bounds, e.g. that the radial component is always
+// positive. there are also some NaNs in the covariance from the regular
+// propagator.
```

Comment:
No, I did not. However, I do not think that we should specifically test the corner cases like this one. These are the kind of things that people will spend days over wondering what is going on. I would also prefer to keep the commented code in the file as a clear reminder that there is something not working or not quite understood. If we remove this, it will get lost.

Reply:
This is changed in the latest iteration.